brush-core/src/shell.rs

```diff
@@ -1268,6 +1268,30 @@ impl Shell {
         self.env.get(name).map(|(_, var)| var)
     }
 
+    /// Tries to add a global variable to the shell's environment.
+    ///
+    /// # Arguments
+    ///
+    /// * `name` - The name of the variable to add.
+    /// * `var` - The variable contents to add.
+    pub fn add_env_var(&mut self, name: &str, var: ShellVariable) -> Result<(), error::Error> {
+        self.env.add(name, var, env::EnvironmentScope::Global)
+    }
+
+    /// Register a builtin to the shell's environment.
+    ///
+    /// # Arguments
+    ///
+    /// * `name` - The in-shell name of the builtin.
+    /// * `registration` - The registration handle for the builtin.
+    pub fn register_builtin<S: Into<String>>(
```

Comment:
Thanks for adding this; while you're at it, would you mind updating the registration for `brushctl` in `brush-shell` to use this new helper?

https://github.com/reubeno/brush/blob/651c185e4446e3421ba22253ba068262ee058487/brush-shell/src/brushctl.rs#L6-L11